src/spark_manager.py

```diff
@@ -143,16 +151,229 @@ def create_cluster(self,
+                try:
+                    # Delete the existing deployment
+                    self.apps_api.delete_namespaced_deployment(
+                        name=self.master_name, namespace=self.namespace
+                    )
+                    logger.info(
+                        f"Deleted existing Spark master deployment: {self.master_name}"
+                    )
+
+                    # Create new deployment
+                    self.apps_api.create_namespaced_deployment(
+                        namespace=self.namespace, body=deployment
+                    )
+                    logger.info(
+                        f"Recreated Spark master deployment: {self.master_name}"
+                    )
+                except ApiException as delete_error:
+                    logger.error(
+                        f"Error replacing Spark master deployment: {delete_error}"
+                    )
+                    raise
+            else:
+                logger.error(f"Error creating Spark master deployment: {e}")
+                raise
+
+    def _create_master_service(self):
+        """Create a Kubernetes service for the Spark master."""
+        service = {
+            "apiVersion": "v1",
+            "kind": "Service",
+            "metadata": {
+                "name": self.master_name,
+                "namespace": self.namespace,
+                "labels": {
+                    "app": "spark",
+                    "component": "master",
+                    "user": self.username,
+                    "cluster-id": self.cluster_id,
+                },
+            },
+            "spec": {
+                "ports": [
+                    {
+                        "port": self.DEFAULT_MASTER_PORT,
+                        "targetPort": self.DEFAULT_MASTER_PORT,
+                        "name": "spark",
+                    },
+                    {
+                        "port": self.DEFAULT_MASTER_WEBUI_PORT,
+                        "targetPort": self.DEFAULT_MASTER_WEBUI_PORT,
+                        "name": "webui",
+                    },
+                ],
+                "selector": {
+                    "app": "spark",
+                    "component": "master",
+                    "user": self.username,
+                    "cluster-id": self.cluster_id,
+                },
+            },
+        }
+
+        try:
+            # Try to create the service first
+            self.core_api.create_namespaced_service(
+                namespace=self.namespace, body=service
+            )
+            logger.info(f"Created Spark master service: {self.master_name}")
+        except ApiException as e:
+            if e.status == 409:  # Conflict - already exists
+                try:
+                    # Delete the existing service
+                    self.core_api.delete_namespaced_service(
+                        name=self.master_name, namespace=self.namespace
+                    )
```

Comment:
Same question - resuse if exists?

Reply:
I actually attempted to reuse service here - it does has a auto selector to select the correct pods. However, during my local testing, if I don't recreate it, the workers won't be able to communicate with master.

Reply:
So this assumes the old master & pods have been deleted and recreated?

Reply:
yea.

Reply:
yes and no actually, if we just created the service - the auto selector will work and select the correct pod. So it doesn't matter if the master pod is the old one or the new one. It selects the current one.

Reply:
:+1: